invokeai/app/api/routers/images.py

```diff
@@ -228,20 +224,33 @@ async def get_image_urls(
     response_model=PaginatedResults[ImageDTO],
 )
 async def list_images_with_metadata(
-    image_type: ImageType = Query(description="The type of images to list"),
-    image_category: ImageCategory = Query(description="The kind of images to list"),
-    page: int = Query(default=0, description="The page of image metadata to get"),
-    per_page: int = Query(
-        default=10, description="The number of image metadata per page"
+    image_origin: Optional[ResourceOrigin] = Query(
```

Comment:
It will be hard to allow both origin types in commercial - the authorization is inherently different for external and internal and the query may get really complex. Is there a valid use case for fetching both types together?

Reply:
Yes - saving the canvas to the gallery results in an image with an external origin.

The client does the work to create a blob from the canvas, then uploads it. But it needs to appear in the gallery as a result.

This is one of the cases that motivated me to implement the changes.